src/Gaffer/FileSystemPath.cpp

```diff
@@ -252,13 +257,13 @@ bool FileSystemPath::isValid( const IECore::Canceller *canceller ) const
 		return true;
 	}
 
-	const file_type t = symlink_status( path( this->string() ) ).type();
-	return t != status_error && t != file_not_found;
+	const boost::filesystem::file_type t = boost::filesystem::symlink_status( boost::filesystem::path( this->string() ) ).type();
+	return t != boost::filesystem::status_error && t != boost::filesystem::file_not_found;
```

Comment:
Since you need to touch this code anyway, I think we might as well convert to `std::filesystem` here. I believe it's just the time-based stuff in `property()` where we need to stick with `boost` a little longer. If that's right, then I think we can convert most of this file to `std::filesystem`.

Reply:
I took out all uses except the time related parts in https://github.com/GafferHQ/gaffer/pull/5049/commits/0a73bc6a93e4c09b3c2b7add9edb7eff393c92fb. One step closer to nudging `boost::filesystem` out the airlock.